csharp.xUnit/GildedRose/ItemType.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+
+namespace GildedRoseKata;
+
+public enum ItemType
+{
+    Normal,
+    Legendary,
+    Dated,
+    Aging,
+    Conjured
+}
+
+public static class ItemTypes
+{
+    private static readonly (string key, ItemType type)[] Keywords =
+    [
+        ("sulfuras", ItemType.Legendary),
+        ("concert", ItemType.Dated),
+        ("aged", ItemType.Aging),
+        ("conjured", ItemType.Conjured)
+    ];
+
+    public static ItemType Detect(string name)
+    {
+        foreach (var (key, type) in Keywords)
+        {
+            if (name.Contains(key, StringComparison.CurrentCultureIgnoreCase))
+            {
+                return type;
+            }
+        }
+
+        return ItemType.Normal;
+    }
+}
+
+public static class ItemTypeExtensions
+{
+    public static int Degrade(this ItemType type, int quality, int sellIn)
+    {
+        if (type == ItemType.Legendary)
+        {
+            return quality;
+        }
+
```

Comment:
Should we remove this?

```suggestion
        if (type == ItemType.Legendary)
        {
            return quality;
        }

```